js/plugins/mcp/src/client/manager.ts

```diff
@@ -0,0 +1,345 @@
+    return new Promise<void>((resolve, reject) => {
+      this._readyListeners.push({ resolve, reject });
+    });
+  }
+
+  /**
+   * Connects to a single MCP server defined by the provided configuration.
+   * If a server with the same name already exists, it will be disconnected first.
+   * Stores the client and transport references internally. Handles connection errors
+   * by marking the server as disabled.
+   * @param serverName The name to assign to this server connection.
+   * @param config The configuration object for the server.
+   */
+  async connect(serverName: string, config: McpServerConfig) {
+    const existingEntry = this._clients[serverName];
+    if (existingEntry) {
+      try {
+        await existingEntry.disconnect();
+      } catch (e) {
+        existingEntry.disable();
+        this.setError(serverName, {
+          message: `[MCP Manager] Error disconnecting from existing connection for ${serverName}`,
+          detail: `Details: ${e}`,
+        });
+      }
+    }
+
+    logger.info(
+      `[MCP Manager] Connecting to MCP server '${serverName}' in manager '${this.name}'.`
+    );
+    try {
+      const client = new GenkitMcpClient({ ...config, name: serverName });
+      this._clients[serverName] = client;
+    } catch (e) {
+      this.setError(serverName, {
+        message: `[MCP Manager] Error connecting to ${serverName} with config ${config}`,
+        detail: `Details: ${e}`,
+      });
+    }
+  }
+
+  /**
+   * Disconnects the specified MCP server and removes its registration
+   * from this client instance.
+   * @param serverName The name of the server to disconnect.
+   */
+  async disconnect(serverName: string) {
+    const client = this._clients[serverName];
+    logger.info(
+      `[MCP Manager] Disconnecting MCP server '${serverName}' in manager '${this.name}'.`
+    );
+    try {
+      await client.disconnect();
+    } catch (e) {
+      client.disable();
+      this.setError(serverName, {
+        message: `[MCP Manager] Error disconnecting from existing connection for ${serverName}`,
+        detail: `Details: ${e}`,
+      });
+    }
+    delete this._clients[serverName];
+  }
+
+  /**
+   * Temporarily disables a server connection. Closes the underlying transport
+   * but retains the server's configuration. Does nothing if the server is
+   * already disabled.
+   * @param serverName The name of the server to disable.
+   */
+  async disable(serverName: string) {
+    const client = this._clients[serverName];
+    if (client.isEnabled()) {
+      logger.info(
+        `[MCP Manager] Disabling MCP server '${serverName}' in manager '${this.name}'`
+      );
+      client.disable();
+    }
+  }
+
+  /**
+   * Re-enables a previously disabled server connection. Attempts to reconnect
+   * using the stored transport. Does nothing if the server is not disabled.
+   * @param serverName The name of the server to re-enable.
+   */
+  async reenable(serverName: string) {
```

Comment:
Let's just do `enable` as suggested above (I know this was my name originally)

Reply:
Done.